File: MarkLite/Controller/StyleViewController.m
//
//  StyleViewController.m
//  MarkLite
//
//  Created by zhubch on 6/30/16.
//  Copyright © 2016 zhubch. All rights reserved.
//

#import "StyleViewController.h"
#import "Configure.h"

@interface StyleViewController ()
@property (nonatomic,weak) IBOutlet UITableView *table;
@end

@implementation StyleViewController
{
    NSArray *styles;
    NSInteger selectedRow;
}

- (void)viewDidLoad {
    [super viewDidLoad];

    self.title = @"渲染样式";
    styles =  @[@"Clearness",@"Clearness Dark",@"GitHub",@"GitHub2",@"Solarized Dark",@"Solarized Light"];
    for (int i = 0; i < styles.count; i++) {
        if ([[Configure sharedConfigure].style isEqualToString:styles[i]]) {
            selectedRow = i;
        }
    }
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return styles.count;
}

- (UITableViewCell*)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [[UITableViewCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@""];
    cell.textLabel.text = styles[indexPath.row];
    if (indexPath.row == selectedRow) {
        UIImageView *imageView = [[UIImageView alloc]initWithFrame:CGRectMake(self.view.bounds.size.width - 40, 5, 30, 30)];
        imageView.image = [UIImage imageNamed:@"check"];
        [cell addSubview:imageView];
    }
    return cell;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return 35;
}

- (CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSInteger)section
{
    return 0.1;
}

- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section
{
    return 15;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    selectedRow = indexPath.row;
    [Configure sharedConfigure].style = styles[selectedRow];
    [self.table reloadData];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: MarkLite/ViewController/StyleViewController.swift
//
//  StyleViewController.swift
//  MarkLite
//
//  Created by zhubch on 2017/8/9.
//  Copyright © 2017年 zhubch. All rights reserved.
//

import UIKit
import RxSwift

class StyleViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
    
    var styles: [String]!
    let table = UITableView(frame: CGRect(), style: .grouped)
    let disposeBag = DisposeBag()

    override func viewDidLoad() {
        super.viewDidLoad()
        
        title = "渲染样式"
        let path = documentPath + "/style/markdown-style/"

        guard let subPaths = FileManager.default.subpaths(atPath: path) else { return }
        
        styles = subPaths.map{ $0.replacingOccurrences(of: ".css", with: "")}.filter{!$0.hasPrefix(".")}
        
        table.rowHeight = 48
        table.delegate = self
        table.dataSource = self
        table.setSeparatorColor(.primary)
        view.addSubview(table)
        
        table.setBackgroundColor(.tableBackground)
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        
        table.frame = self.view.bounds
    }
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return styles.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = UITableViewCell(style: .default, reuseIdentifier: nil)
        cell.textLabel?.text = styles[indexPath.row]
        cell.textLabel?.setTextColor(.primary)
        cell.textLabel?.font = UIFont.font(ofSize: 16)
        cell.setBackgroundColor(.background)
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        Configure.shared.markdownStyle.value = styles[indexPath.row]
        tableView.deselectRow(at: indexPath, animated: true)
    }
    
    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
        return 0.01
    }
    
    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
        return 20
    }

}
